test/cppsuite/test_harness/workload/database_operation.cpp

```diff
@@ -246,6 +246,75 @@ database_operation::read_operation(thread_context *tc)
         tc->transaction.rollback();
 }
 
+void
+database_operation::remove_operation(thread_context *tc)
+{
+    logger::log_msg(
+      LOG_INFO, type_string(tc->type) + " thread {" + std::to_string(tc->id) + "} commencing.");
+
+    /*
+     * We need two types of cursors. One cursor is a random cursor to randomly select a key and the
+     * other one is a standard cursor to remove the random key. This is required as the random
+     * cursor does not support the remove operation.
+     */
+    std::map<uint64_t, scoped_cursor> rnd_cursors, cursors;
+
+    /* Loop while the test is running. */
+    while (tc->running()) {
+        /*
+         * Sleep the period defined by the op_rate in the configuration. Do this at the start of the
+         * loop as it could be skipped by a subsequent continue call.
+         */
+        tc->sleep();
+
+        /* Choose a random collection to update. */
+        collection &coll = tc->db.get_random_collection();
+
+        /* Look for existing cursors in our cursor cache. */
+        if (cursors.find(coll.id) == cursors.end()) {
+            logger::log_msg(LOG_TRACE,
+              "Thread {" + std::to_string(tc->id) +
+                "} Creating cursor for collection: " + coll.name);
+            /* Open the two cursors for the chosen collection. */
+            scoped_cursor rnd_cursor =
+              tc->session.open_scoped_cursor(coll.name, "next_random=true");
+            rnd_cursors.emplace(coll.id, std::move(rnd_cursor));
+            scoped_cursor cursor = tc->session.open_scoped_cursor(coll.name);
+            cursors.emplace(coll.id, std::move(cursor));
+        }
+
+        /* Start a transaction if possible. */
+        tc->transaction.try_begin();
+
+        /* Get the cursor associated with the collection. */
+        scoped_cursor &rnd_cursor = rnd_cursors[coll.id];
+        scoped_cursor &cursor = cursors[coll.id];
+
+        /* Choose a random key to delete. */
+        const char *key_str;
+        int ret = rnd_cursor->next(rnd_cursor.get());
+        /* It is possible not to find anything if the collection is empty. */
```

Comment:
Is this going to blow up our update_operation? I.e. if we try and update a removed record, or does WiredTiger allow updates to become inserts?

Reply:
Before calling `update` on a key, we could check if it exists to make sure we are updating it. If it does not exist, we could call `insert` in the `update_operation` or just get another key.
```
        /* Choose a random key to update. */
        uint64_t key_id =
          random_generator::instance().generate_integer<uint64_t>(0, coll.get_key_count() - 1);
        if(key_id.exists())
            if (!tc->update(cursor, coll.id, tc->key_to_string(key_id))) {
                tc->transaction.rollback();
            }
       else
            if (!tc->insert(cursor, coll.id, tc->key_to_string(key_id))) {
                tc->transaction.rollback();
            }
```

Reply:
Yes, WiredTiger allows updates to become inserts. If the record does not exist and we call `update`, it becomes an insert. Hence the suggestion I made just above is not worth it.

I had to finish the transaction of the remove thread if no records is found so we can get a chance to find newer ones, see bef13fdfb943f0ce6888dd6f59a5fbf3eccffb1a.